### PR TITLE
New formula libfreenect2

### PR DESCRIPTION
**Pull request by xionglingfeng**
New formula `libfreenect2`. Some dirty hack applied to workaround a strange `cmake` [bug](https://github.com/Homebrew/homebrew/issues/41647).

### CODE REVIEW

File: libfreenect2.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
+require "formula"
+
+class Libfreenect2 < Formula
+  desc "Drivers and libraries for the Xbox Kinect device version 2"
+  homepage "http://openkinect.org"
+  #url "https://github.com/OpenKinect/libfreenect/archive/v0.5.1.tar.gz"
```
Comment (dunn) [inline libfreenect2.rb]: These two lines are from the stable version so they can be removed.

File: libfreenect2.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
+require "formula"
```
Comment (DomT4) [inline libfreenect2.rb]: This hasn't been required for a while now.

File: libfreenect2.rb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+  depends_on "cmake" => :build
+  depends_on "pkg-config" => :build
+  depends_on "libusb"
+  depends_on "jpeg-turbo"
+
+  def install
+    args = std_cmake_args
+
+    inreplace "examples/protonect/CMakeLists.txt", /  \$\{LibUSB_LIBRARIES\}/, "usb-1.0"
```
Comment (DomT4) [inline libfreenect2.rb]: I suspect audit is going to tell you to wrap the regex in `%r[]` rather than `//`

File: libfreenect2.rb [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+    if build.universal?
+      ENV.universal_binary
+      args << "-DCMAKE_OSX_ARCHITECTURES=#{Hardware::CPU.universal_archs.as_cmake_arch_flags}"
+    end
+
+    mkdir "build" do
+      args << "-DLibUSB_INCLUDE_DIRS=#{Formula["libusb"].include}/libusb-1.0"
+      system "cmake", "../examples/protonect", *args
+      system "make install"
```
Comment (DomT4) [inline libfreenect2.rb]: `system "make", "install"`

**Comment by xionglingfeng** [comment]
@DomT4 I have updated the formula as you requested. Thank you.

File: libfreenect2.rb [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+  homepage "http://openkinect.org"
+  head "https://github.com/OpenKinect/libfreenect2.git"
+
+  option :universal
+
+  depends_on "cmake" => :build
+  depends_on "pkg-config" => :build
+  depends_on "libusb"
+  depends_on "jpeg-turbo"
+  depends_on "glfw3"
```
Comment (DomT4) [inline libfreenect2.rb]: You'll need the direct path here `depends_on "homebrew/versions/glfw3"`.

**Comment by ahundt** [comment]
@xionglingfeng could you make that last change? I'd love to use this

**Comment by xionglingfeng** [comment]
@DomT4 @ahundt updated.

**Comment by ahundt** [comment]
@xionglingfeng seems to have failed the checks

**Comment by xionglingfeng** [comment]
@ahundt  please check again. It should work for now.

**Comment by ahundt** [comment]
It seems the bot test failed?

File: libfreenect2.rb (hunk 5 above)
Comment (RyanGordon) [inline libfreenect2.rb]: Just curious, why is the direct path needed?

**Comment by dunn** [comment]
Unfortunately, since this PR was opened we've stopped adding new formula to the head-only tap.  There are two release candidates now, so if you'd like you can submit it to devel-only.  Thanks!

**Pull request closed without merging**